python/lsst/rubintv/production/pipelineRunning.py

```diff
@@ -256,62 +309,70 @@ def callback(self, payload):
                 # which does the decoding too to provide a unified interface.
-                if "visit" in payload.dataId:
-                    visit = f"{payload.dataId['visit']}"
-                else:
-                    visit = f"{payload.dataId['exposure']}"
-                self.watcher.redisHelper.redis.lpush(f"{self.instrument}-PSFPLOTTER", visit)
+                if who == "SFM":
+                    self.redisHelper.redis.lpush(f"{self.instrument}-PSFPLOTTER", compoundId)
             if self.step == "nightlyRollup":
-                self.watcher.redisHelper.reportNightLevelFinished(self.instrument)
+                self.redisHelper.reportNightLevelFinished(self.instrument, who=who)
 
         except Exception as e:
             if self.step == "step2a":
-                self.watcher.redisHelper.reportVisitLevelFinished(self.instrument, "step2a", failed=True)
+                self.redisHelper.reportVisitLevelFinished(self.instrument, "step2a", who=who, failed=True)
             if self.step == "nightlyRollup":
-                self.watcher.redisHelper.reportNightLevelFinished(self.instrument, failed=True)
+                self.redisHelper.reportNightLevelFinished(self.instrument, who=who, failed=True)
             raiseIf(self.doRaise, e, self.log)
 
-    def postProcessQuantum(self, quantum, processingId):
+    def postProcessQuantum(self, quantum) -> None:
         """Write shards here, make sure to keep these bits quick!
 
+        compoundId is a maybe-compound id, either a single exposure or a
+        compound of multiple exposures, depending on the pipeline, joined with
+        a "+".
+
         Also, anything you self.limitedButler.get() make sure to add to
         cache_on_put.
 
         # TODO: After OR3, move all this out to postprocessQuanta.py, and do
         the dispatch and function definitions in there to keep the runner
         clean.
         """
-        match quantum.taskName:
-            case "lsst.ip.isr.isrTask.IsrTask":
-                self.postProcessIsr(quantum)
-            case "lsst.pipe.tasks.calibrate.CalibrateTask":
-                # TODO: think about if we could make dicts of some of the
-                # per-CCD quantities like PSF size and 50 sigma source counts
-                # etc. Would probably mean changes to mergeShardsAndUpload in
-                # order to merge dict-like items into their corresponding
-                # dicts.
-
-                self.postProcessCalibrate(quantum, processingId)
-            case "lsst.pipe.tasks.postprocess.ConsolidateVisitSummaryTask":
-                # ConsolidateVisitSummaryTask regardless of quickLook or NV
-                # pipeline, because this is the quantum that holds the
-                # visitSummary
-                self.postProcessVisitSummary(quantum)
-            case _:
-                return
-
-    def postProcessIsr(self, quantum):
+        taskName = quantum.taskName
```

Comment:
Curious about your opinion on this from a design perspective actually: I went to pull all these out, but `self` appears in so many of these that they don't really suit being free-floating function as they need `LocationConfigs` and all sorts of other state. So I could make a new `QuantumPostProcessor` class with those things on it, and have the runner have on on the class, and just call that for each quantum, but then I'm kind of curious if I'm actually gaining anything tbh. What do you think?

Reply:
I agree that you don't to gain much by just moving them out of this class.  Ideally they'd have some abstract interface and be configured wherever you configure the pipeline rather than dispatched by task class name, but that's mostly about extensibility in a direction I don't think you care much about now (i.e. making more of this pipeline-agnostic).